### PR TITLE
updates @okta/okta-auth-js to 4.0.0

### PR DESCRIPTION
**Pull request by aarongranick-okta**
This is for SIW 5.0.0 release

## Description:
Also updates unit tests from AMD to ES6
tokenmanager.autoRenew is forcibly disabled. This will prevent a background process from being started which will attempt to auto renew tokens. Running this process in the SIW may interfere with background renew processes in other javascript SDKs.

## PR Checklist

- [ ] Have you verified the basic functionality for this change?

- [ ] Added unit tests?

- [ ] Added e2e tests

- [ ] Did you follow our [Security Best Practices](https://oktawiki.atlassian.net/wiki/display/eng/Security+Best+practices)?

### Issue:

- [OKTA-324596](https://oktainc.atlassian.net/browse/OKTA-324596)

### CODE REVIEW

File: src/widget/createAuthClient.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
+var OktaAuth = require('@okta/okta-auth-js').OktaAuth;
+var Util = require('util/Util');
+var config   = require('config/config.json');
```
Comment (haishengwu-okta) [inline src/widget/createAuthClient.js:3]: nit: remove extra space

File: src/widget/OktaSignIn.js [hunk 2]
```diff
@@ -130,7 +110,16 @@ var OktaSignIn = (function () {
         See: https://developer.okta.com/code/javascript/okta_sign-in_widget#cdn
       `);
 
-    var authClient = createAuthClient(options);
+    var authParams = _.extend({
```
Comment (haishengwu-okta) [inline src/widget/OktaSignIn.js:113]: why not leave all those extra logic to `createAuthClient` function?

Comment (aarongranick-okta) [reply src/widget/OktaSignIn.js:113]: The `createAuthClient` is not bound to a widget instance, so it doesn't have access to `this.options` (it is a factory function, also used by tests, when there is no widget instance). This block of code is copying specific properties from `this.options` onto `authParams`

Comment (aarongranick-okta) [reply src/widget/OktaSignIn.js:113]: Basically, the old function took WIDGET options, the new function takes AUTHJS options.

**Review by haishengwu-okta — APPROVED** [review]

**Pull request closed without merging**